release/ray_release/test_automation/ci_state_machine.py

```diff
@@ -0,0 +1,42 @@
+from ray_release.test_automation.state_machine import TestStateMachine
+
+from ray_release.test import Test, TestState
+
+
+class CITestStateMachine(TestStateMachine):
+    def _move_hook(self, from_state: TestState, to_state: TestState) -> None:
+        change = (from_state, to_state)
+        if change == (TestState.PASSING, TestState.CONSITENTLY_FAILING):
+            self._create_github_issue()
+        elif change == (TestState.FAILING, TestState.CONSITENTLY_FAILING):
+            self._create_github_issue()
+        elif change == (TestState.CONSITENTLY_FAILING, TestState.PASSING):
+            self._close_github_issue()
+
+    def _state_hook(self, state: TestState) -> None:
+        pass
+
+    def _create_github_issue(self) -> None:
+        labels = [
+            "P0",
+            "bug",
+            "ci-test",
+            "ray-test-bot",
+            "triage",
+            self.test.get_oncall(),
+        ]
+        issue_number = self.ray_repo.create_issue(
+            title=f"CI test {self.test.get_name()} failed",
+            body=(
+                f"CI test **{self.test.get_name()}** failed. "
+                f"See {self.test_results[0].url} for more details.\n\n"
+                f"Managed by OSS Test Policy"
+            ),
+            labels=labels,
+            assignee="can-anyscale",
```

Comment:
Totally, I worry it will be spammy (so many flaky tests) so want it go to me first before triaging further; when it's stable i'll find a way to for it go directly to the oncall